diff --git a/server/src/repositories/media.repository.ts b/server/src/repositories/media.repository.ts
index 90570873510a1780a918840025e121f2160fcb31..012bc3637ac2ded0269d7e77ce363b136381383c 100644
--- a/server/src/repositories/media.repository.ts
+++ b/server/src/repositories/media.repository.ts
@@ -5,6 +5,7 @@ import { camelCase, upperFirst } from 'lodash-es';
 import { Duration } from 'luxon';
 import { spawn } from 'node:child_process';
 import fs from 'node:fs/promises';
+import { devNull } from 'node:os';
 import { Writable } from 'node:stream';
 import sharp, { Sharp } from 'sharp';
 import type {
@@ -412,7 +413,7 @@ export class MediaRepository {
     // recommended for vp9 for better quality and compression
     return new Promise((resolve, reject) => {
       // first pass output is not saved as only the .log file is needed
-      this.configureFfmpegCall(input, '/dev/null', options)
+      this.configureFfmpegCall(input, devNull, options)
         .addOptions('-pass', '1')
         .addOptions('-passlogfile', output)
         .addOptions('-f null')
