diff --git a/server/src/dtos/env.dto.ts b/server/src/dtos/env.dto.ts
index 71658a6bf..d682d3fba 100644
--- a/server/src/dtos/env.dto.ts
+++ b/server/src/dtos/env.dto.ts
@@ -1,3 +1,4 @@
+import path from 'node:path';
 import z from 'zod';
 import { ImmichEnvironmentSchema, LogFormatSchema, LogLevelSchema } from 'src/enum.js';
 import { IsIPRange } from 'src/validation.js';
@@ -12,7 +13,7 @@ enum DatabaseSslMode {
 }
 
 const DatabaseSslModeSchema = z.enum(DatabaseSslMode).describe('Database SSL mode').meta({ id: 'DatabaseSslMode' });
-const absolutePath = z.string().regex(/^\//, 'Must be an absolute path').optional();
+const absolutePath = z.string().refine((value) => path.isAbsolute(value), 'Must be an absolute path').optional();
 /**
  * Treat certain strings as booleans and coerce them to boolean
  * Ideal for environment variables that are strings but should be treated as booleans
