diff --git a/server/src/cores/storage.core.ts b/server/src/cores/storage.core.ts
index fa9d21a06c72b7f8361c220f503c1ce3d53f7a33..13e9bdc360550c7917ba101c987956692032bdc3 100644
--- a/server/src/cores/storage.core.ts
+++ b/server/src/cores/storage.core.ts
@@ -1,4 +1,4 @@
-import { dirname, join, resolve } from 'node:path';
+import { dirname, isAbsolute, join, relative, resolve, sep } from 'node:path';
 import type { VideoInterfaces } from 'src/types.js';
 import { StorageAsset } from 'src/database.js';
 import {
@@ -151,12 +151,13 @@ export class StorageCore {
 
   static isImmichPath(path: string) {
     const resolvedPath = resolve(path);
-    const resolvedAppMediaLocation = StorageCore.getMediaLocation();
-    const normalizedPath = resolvedPath.endsWith('/') ? resolvedPath : resolvedPath + '/';
-    const normalizedAppMediaLocation = resolvedAppMediaLocation.endsWith('/')
-      ? resolvedAppMediaLocation
-      : resolvedAppMediaLocation + '/';
-    return normalizedPath.startsWith(normalizedAppMediaLocation);
+    const resolvedAppMediaLocation = resolve(StorageCore.getMediaLocation());
+    const relativePath = relative(resolvedAppMediaLocation, resolvedPath);
+
+    return (
+      relativePath === '' ||
+      (relativePath !== '..' && !relativePath.startsWith(`..${sep}`) && !isAbsolute(relativePath))
+    );
   }
 
   async moveAssetImage(asset: StorageAsset, fileType: AssetFileType, format: ImageFormat) {
