diff --git a/server/src/repositories/storage.repository.ts b/server/src/repositories/storage.repository.ts
index eaed895531ab0e1a47087425e4b3275a8a9cd82e..9d0800de7abe51416b2aec7148bece1040e7214a 100644
--- a/server/src/repositories/storage.repository.ts
+++ b/server/src/repositories/storage.repository.ts
@@ -300,7 +300,7 @@ export class StorageRepository {
   private asGlob(pathToCrawl: string): string {
     // eslint-disable-next-line import-x/no-named-as-default-member
     const escapedPath = fastGlob
-      .escapePath(pathToCrawl)
+      .convertPathToPattern(pathToCrawl)
       .replaceAll('"', '["]')
       .replaceAll("'", "[']")
       .replaceAll('`', '[`]');
